pkg/operator/discover/discover_test.go

```diff
@@ -118,10 +118,33 @@ func TestGetAvailableDevices(t *testing.T) {
 	devices, err := GetAvailableDevices(context, nodeName, ns, d, "^sd.", false)
 	assert.Nil(t, err)
 	assert.Equal(t, 1, len(devices))
+	// devices should be in used now, 2nd try gets empty devices
+	devices, err = GetAvailableDevices(context, nodeName, ns, d, "^sd.", false)
+	assert.Nil(t, err)
+	assert.Equal(t, 0, len(devices))
+
+	err = FreeDevices(context, nodeName, ns)
+	assert.Nil(t, err)
+	// all devices freed
 	devices, err = GetAvailableDevices(context, nodeName, ns, nil, "^sd.", false)
 	assert.Nil(t, err)
 	assert.Equal(t, 4, len(devices))
+	// devices should be in used now, 2nd try gets empty devices
+	devices, err = GetAvailableDevices(context, nodeName, ns, nil, "^sd.", false)
+	assert.Nil(t, err)
+	assert.Equal(t, 0, len(devices))
+
+	err = FreeDevices(context, nodeName, ns)
+	assert.Nil(t, err)
+
 	devices, err = GetAvailableDevices(context, nodeName, ns, nil, "", true)
 	assert.Nil(t, err)
 	assert.Equal(t, 5, len(devices))
+	// devices should be in used now, 2nd try gets empty devices
+	devices, err = GetAvailableDevices(context, nodeName, ns, nil, "", true)
+	assert.Nil(t, err)
+	assert.Equal(t, 0, len(devices))
+
+	err = FreeDevices(context, nodeName, ns)
+	assert.Nil(t, err)
```

Comment:
Would it make sense to "double" check if the ConfigMap is now empty too?

Reply:
if the configmap is not deleted, err should be raised and assert fail here